h/claim/__init__.py

```diff
@@ -0,0 +1,13 @@
+from webob.cookies import SignedSerializer
+
+
+def includeme(config):
+    config.include('.views')
+    config.include('.util')
+
+    # We use the shared session secret, but salt it with the namespace
+    # 'h.claim' -- only messages serialized with this salt will
+    # authenticate on deserialization.
+    secret = config.registry.settings['session.secret']
+    serializer = SignedSerializer(secret, 'h.claim')
+    config.registry.claim_serializer = serializer
```

Comment:
The re-use of the session secret, even with a salt, makes me a bit nervous. I'm not an expert in this department, of course. However, the difference between this and the session is that since there is nothing else in the payload other than the user id an attacker could maybe register lots of users for email addresses they own and perhaps execute a chosen plaintext attack, whereas with the session there is the csrf token in there that an attacker cannot select.

Maybe it'd be enough to just stick a random number inside the payload to make it impossible to select the plaintext.